Update project-variables.md (#5387)
dbt-labs/docs.getdbt.com: website/docs/docs/build/project-variables.md

</VersionBlock>

> If dbt is unable to find a definition for a variable after checking these five places, then a compilation error will be raised.

Reviewer: It looks like we still list 4 places for 1.5 (it changes when you select the version in the version picker), so suggesting we be more specific:
```suggestion
If dbt is unable to find a definition for a variable after checking all possible variable declaration places, then a compilation error will be raised.
```
